File: ui/src/types/configmap.ts
import type { ObjectMeta } from "./shared";

export interface CreateConfigMapRequest {
  name: string;
  namespace: string;

  labels?: Record<string, string>;
  annotations?: Record<string, string>;

  data?: Record<string, string>;
  binaryData?: Record<string, string>; // base64-encoded strings over the wire

  immutable?: boolean;
}

export interface ConfigMap {
  metadata: ObjectMeta;
  data?: Record<string, string>;
  binaryData?: Record<string, string>;
  immutable?: boolean;
}

export interface ConfigMapList {
  items: ConfigMap[];
}

File: backend/internal/builders/configmap.go
package builders

import (
	"github.com/podforge/backend/internal/types"
	corev1 "k8s.io/api/core/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

func BuildConfigmap(req types.CreateConfigmapRequest) *corev1.ConfigMap {
	return &corev1.ConfigMap{
		ObjectMeta: metav1.ObjectMeta{
			Name:        req.Name,
			Namespace:   req.Namespace,
			Labels:      req.Labels,
			Annotations: req.Annotations,
		},
		Immutable:  req.Immutable,
		Data:       req.Data,
		BinaryData: req.BinaryData,
	}
}
